### DISCUSSION TITLE
Ensure deterministic placement of lovelace cards and allow hints for card height

### DISCUSSION THREAD

**Discussion by R5fan** [home-assistant/frontend#8165] [Dashboard]
Every time I reload my 3 columned lovelace front end, I end up with a different layout. The placement of the cards is not deterministic. More over, the layout is often completely illogical with some columns being more than twice as long as others. Example:
![image](https://user-images.githubusercontent.com/1439530/104714497-90391a80-5725-11eb-9320-8bdaab0a808e.png)

Im not completely sure what causes the randomness, but i think that lovelace has problems predicting the height of certain cards before they are rendered, and thus organizes them, well, not very intelligently. AFAICT this is at least true for image cards / camera views and it is definitely the case for conditional cards, where it seems to assume the conditions are always met (I have a ton of conditional cards below the media control buttons in the third column). It may affect other cards too.

here is my proposed solution: allow us to give LL hints about the height of the card, so that LL can use that hint to determine their placement. That should make it both deterministic and allow us more flexibility in choosing the placement, particularly prevent conditional cards that are rarely shown from messing up our layouts.

**Comment by spacegaier** [comment]
If you put your cards in stacks / grids the will stay and not be moved to other columns.

**Reply by R5fan** [reply]
I tried grids, but its too inflexible, especially when the frontend is work in progress. It also doesnt seem to work with conditional cards.

Vertical stacks isnt really a solution either, because my HA tablet shows 3 columns (landscape). my phone shows 1 in portait and 2 in landscape and my desktop shows 4.  Making different layouts for all of them is just silly work. 

Just an optional card height hint would seem like a much easier and neater solution.
